src/Shared/Components/ProtectedPrerenderComponentApplicationStore.cs

```diff
@@ -20,15 +23,17 @@ public ProtectedPrerenderComponentApplicationStore(IDataProtectionProvider dataP
         public ProtectedPrerenderComponentApplicationStore(string existingState, IDataProtectionProvider dataProtectionProvider)
         {
             CreateProtector(dataProtectionProvider);
-            ExistingState = JsonSerializer.Deserialize<Dictionary<string, byte[]>>(_protector.Unprotect(Convert.FromBase64String(existingState)));
+            DeserializeState(_protector.Unprotect(Convert.FromBase64String(existingState)));
         }
 
-        protected override byte[] SerializeState(IReadOnlyDictionary<string, byte[]> state)
+        protected override PooledByteBufferWriter SerializeState(IReadOnlyDictionary<string, ReadOnlySequence<byte>> state)
         {
             var bytes = base.SerializeState(state);
             if (_protector != null)
             {
-                bytes = _protector.Protect(bytes);
+                var newBuffer = new PooledByteBufferWriter(_protector.Protect(bytes.WrittenMemory.Span.ToArray()));
```

Comment:
Wait, is this the real usage?

Reply:
This is until we get spanified data protection APIs for it :(. If you can think of a way around it, I'm happy to look into it.